ewc-rfc004-individual-wallet-attestation.md

```diff
@@ -0,0 +1,146 @@
+Furthermore, an attestation provider must be able to determine if a wallet instance offers functional capabilities and meets the security requirements for issuance. For this purpose, the attestation provider must receive an attestation of the wallet’s capabilities and proof of non-revocation for the wallet instance.
+
+# 3.0	Wallet Trust Evidence (WTE)
+
+The attestation of a wallet's capabilities through Wallet Trust Evidence is based on the OAuth 2.0 Attestation-Based Client Authentication draft 03 [1].
+
+## 3.1 Wallet Trust Evidence JWT
+
+The following Wallet Trust Evidence (WTE) JWT must be presented when sending the token request as defined in section 6.1 of OID4VCI[2].
+
+```
+{
+        "typ": "wallet-trust-evidence+jwt",
+        "alg": "ES256",
+        "kid": "1"
+}
+.
+{
+        "iss": "https://attestation-service.com",
+        "sub": "https://wallet-solution.com",
+        "iat": 1541493724,
+        "status": {
+          "status_list": {
+            "idx": 0,
+            "uri": "https://example.com/statuslists/1"
+          },
+        },
+        "attested_security_context" : "https://eudiwalletconsortium.org/",
+        "cnf": {
+                "jwk" : {
+                        "kty": "EC",
+                        "crv": "P-256",
+                        "x": "TCAER19Zvu3OHF4j4W4vfSVoHIP1ILilDls7vCeGemc",
+                        "y": "ZxjiWWbZMQGHVWKVQ4hbSIirsVfuecCE6t4jT9F2HZQ"
+                }
+        }
+}
+```
+
+
+This profile defines the following additional requirements for a Client Attestation JWT from [1] used as a WTE JWT in this context.
+
+1. The alg used to sign the attestation must be “ES256”
+2. The kid must identify a JWK which must be resolvable as a JWK Set [https://datatracker.ietf.org/doc/html/rfc7517#section-5] by appending ./well-known/jwt-issuer to the value of the iss claim
+3. The iss claim must be the unique URL of the wallet solutions provider’s attestation service
+4. All acceptable URLs for “iss” claim must be registered within this profile’s section 5.
+5. The sub claim must be the qualified client_id of the wallet solution and registered within section 5 of this profile.
+6. The attestation must include a status claim which contains a reference to a status list as defined in [3] and which allows the attestation provider to check the state of revocation of the wallet instance
+7. The value of attested_security_context must be “https://eudiwalletconsortium.org/”
+8. The cnf claim must contain a Json Web Key (JWK) which must be used as for Proof of Possession (PoP) as defined in [1]
+
+### 3.1.1 Wallet Trust Evidence PoP JWT
+
+A Wallet Trust Evidence JWT must be provided as defined by section 4.3 of Attestation-based Client Authentication[1].
+
+# 4.0	Messages
```

Comment:
@lalc, @georgepadayatti, @ntsbs Not sure though if we can use RFC001 for getting the WTE. The flow should start whenever the wallet is opened the first time and triggered by the wallet towards the wallet provider. Next, the wallet provider requests information about the device where the wallet is running on. These data are collected, send to the wallet provider, which verifies those and if they are okay, these data set is signed and returned to the wallet instance. Not sure if this complex flow can be handled by RFC001.

Reply:
> Not sure if this complex flow can be handled by RFC001

Due to that and because this is dependent on the wallet solution architecture (e.g. a cloud wallet will handle this differently than a native edge wallet) it does not make sense to specify that here

RFC001 is not suited for the issuance of the Wallet Unit Attestation because the issuer has no means of authenticating the wallet other than using a wallet attestation. So you would end up in a chicken-egg problem.